x/gamm/keeper/pool.go

```diff
@@ -120,14 +125,20 @@ func (k Keeper) GetPoolsAndPoke(ctx sdk.Context) (res []types.TraditionalAmmInte
 	for ; iter.Valid(); iter.Next() {
 		bz := iter.Value()
 
-		pool, err := k.UnmarshalPool(bz)
+		poolI, err := k.UnmarshalPool(bz)
 		if err != nil {
 			return nil, err
 		}
 
-		if pokePool, ok := pool.(types.WeightedPoolExtension); ok {
+		if pokePool, ok := poolI.(types.WeightedPoolExtension); ok {
 			pokePool.PokePool(ctx.BlockTime())
 		}
+
+		pool, ok := poolI.(types.TraditionalAmmInterface)
+		if !ok {
+			return nil, fmt.Errorf("type cast failed")
```

Comment:
ditto for error. Maybe we could define this error in x/gamm/types/errors.go and reuse where applicable?

Reply:
Same question as above